deployment/deployment.py

```diff
@@ -0,0 +1,267 @@
+import logging
+import os
+import time
+
+from common_ci_utils.command_runner import exec_cmd
+from common_ci_utils.exceptions import (
+    AggregateNodeStatusCheckFailed,
+    StorageStatusCheckFailed,
+)
+from common_ci_utils.file_system_utils import create_directory, set_permissions
+from common_ci_utils.host_info import get_ip_address
+from common_ci_utils.package_fetcher import download_rpm
+from common_ci_utils.postgres_utils import enable_postgresql_version
+from common_ci_utils.rpm_manager import install_rpm
+from common_ci_utils.templating import Templating
+from deployment.npm import NPM
+from framework import config
+
+log = logging.getLogger(__name__)
+
+
+class Deployment(object):
+    """
+    Noobaa SA deployment
+    """
+
+    def __init__(self):
+        """
+        Initializes the necessary variables needed for
+        Noobaa Standalone Deployment
+        """
+        self.rpm_url = config.ENV_DATA["noobaa_sa"]
+        self.postgres_repo = config.ENV_DATA["postgres_repo"]
+        self.packages = config.ENV_DATA["db_packages"]
+        self.postgresql_version = config.ENV_DATA["postgresql_version"]
+        self.package = config.ENV_DATA["package_json"]
+        self.npm = NPM(self.package)
+        config.ENV_DATA["ip_address"] = get_ip_address()
+        self.sleep = 10
+
+    def install_noobaa_sa(self):
+        """
+        Installs Noobaa Standalone deployment
+        """
+        log.info("Installing Noobaa Standalone")
+        # install noobaa standalone rpm
+        rpm_path = download_rpm(rpm_url=self.rpm_url)
+        install_rpm(rpm_path=rpm_path)
+
+        # enable postgres repo
+        install_rpm(rpm_path=self.postgres_repo)
+
+        # enable postgresql version to default
+        enable_postgresql_version(self.postgresql_version)
+
+        # install postgresql
+        install_rpm(packages=self.packages)
+
+        # set permissions
+        noobaa_core_dir = config.ENV_DATA["noobaa_core_dir"]
+        set_permissions(
+            directory_path=noobaa_core_dir,
+            permissions=777,
+            use_sudo=True
+        )
+
+        # create storage directory
+        previous_dir = os.getcwd()
+        storage_dir = config.ENV_DATA["storage_dir"]
+        noobaa_core_dir = config.ENV_DATA["noobaa_core_dir"]
+        os.chdir(noobaa_core_dir)
```

Comment:
I would propose to change the dir to `noobaa_core_dir` inside each method where it is required (`initialize_db`, `run_db`, `create_db`,...) so it will be possible to call those methods later from anywhere without taking care of changing the working directory. (It might be done via decorator maybe or inside the `npm.run_script` method.)

Reply:
I will take up in follow up PR